data/types/device/mode/mode.go

```diff
@@ -0,0 +1,82 @@
+package mode
+
+import (
+	"github.com/tidepool-org/platform/data"
+	commontypes "github.com/tidepool-org/platform/data/types/common"
+	"github.com/tidepool-org/platform/data/types/device"
+	"github.com/tidepool-org/platform/structure"
+	structureValidator "github.com/tidepool-org/platform/structure/validator"
+)
+
+const (
+	ConfidentialMode = "confidential"
+	ZenMode          = "zen"
+)
+
+func Modes() []string {
+	return []string{
+		ConfidentialMode,
+		ZenMode,
+	}
+}
+
+type Mode struct {
+	device.Device `bson:",inline"`
+	EventID       *string               `json:"eventId,omitempty" bson:"eventId,omitempty"`
+	Duration      *commontypes.Duration `json:"duration,omitempty" bson:"duration,omitempty"`
+}
+
+func New(subType string) *Mode {
+	return &Mode{
+		Device: device.New(subType),
+	}
+}
+
+func NewWithEvent(subType string, deviceEvent *string) *Mode {
+	return &Mode{
+		Device: device.NewWithEvent(subType, deviceEvent),
+	}
+}
+
+func (m *Mode) Parse(parser structure.ObjectParser) {
+	if !parser.HasMeta() {
+		parser = parser.WithMeta(m.Meta())
+	}
+
+	m.Device.Parse(parser)
+	m.EventID = parser.String("eventId")
+	m.Duration = commontypes.ParseDuration(parser.WithReferenceObjectParser("duration"))
+}
+
+func (m *Mode) Validate(validator structure.Validator) {
+	if !validator.HasMeta() {
+		validator = validator.WithMeta(m.Meta())
+	}
+
+	m.Device.Validate(validator)
+	validator.String("subType", &m.SubType).OneOf(Modes()...)
+	if m.EventType != nil {
+		validator.String("eventType", m.EventType).Exists()
+	}
```

Comment:
```suggestion
	validator.String("eventType", m.EventType).Exists()
```
eventType should be mandatory.